util/Logger.h

```diff
@@ -1,56 +1,245 @@
+    error) to filter which log records are generated at the sources and which
+    log records are written at the sinks. Logs that are filtered out at either
+    the sink or the source are not generated by the source.
+
+    The intended uses of the levels are:
+    error - used for "major" unrecoverable errors which will affect game play.  Error level issues
+            need to be fixed.  Error level will probably not be turned off unless they are flooding
+            the logs.
+            Examples are: the game is about to crash, a string is missing from the stringtable, etc.
+    warn  - used for "minor", recoverable errors that will not affect game play but do indicate a
+            problem.
+            For example a missing id that can be ignored, an extra item in a container.
+    info  - used to report normal game state and progress.  This should be the default level of
+            logging.  The number of log should be low enought to not floow the logs.
+            For example reporting that the network connected/disconnected.
+    debug - used for low-level implementation or calculation details.  For a named logger this
+            level will probably only be turned on by devs working on that
+            section of code.  This will be detailed and perhaps voluminous.
+            For example reporting that the network disconnected via a client initiated shutdown
+            with a linger time of 30 ms before closing.
+    trace - used for the highest detail of logging.
+
+    Sources are further filtered by name.  For example "combat" is the name of the combat system's logger.
+
+    Usage:
+
+    The default loggers as stream outputs, like std::cout as follow:
+
+    ErrorLogger() << "Put any streamable output here.  It will only be
+                  << "computed at the error threshold or greater.";
+    WarnLogger()  << "Put any streamable output here.  It will only be
+                  << "computed at the warn threshold or greater.";
+    InfoLogger()  << "Put any streamable output here.  It will only be
+                  << "computed at the info threshold or greater.";
+    DebugLogger() << "Put any streamable output here.  It will only be
+                  << "computed at the debug threshold or greater.";
+    TraceLogger() << "Put any streamable output here.  It will only be
+                  << "computed at the trace threshold or greater.";
+
+    The named loggers are created with:
+    CreateThreadedLogger(name_of_logger);
+
+    And used with
+
+    ErrorLogger(name_of_logger) << "any streamable output, only computed "
+                                << "at the error threshold or greater "
+                                << "for the 'name_of_logger' logger.";
+    WarnLogger(name_of_logger)  << "streamable output";
+    InfoLogger(name_of_logger)  << "streamable output";
+    DebugLogger(name_of_logger) << "streamable output";
+    TraceLogger(name_of_logger) << "streamable output";
+*/
```

Comment:
not clear how to set a minimum log level for a particular logger... which I thought was a thing?